modules/programs/htop.nix

```diff
@@ -0,0 +1,252 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.htop;
+
+  list = xs: concatMapStrings (x: "${toString x} ") xs;
+
+  bool = b: if b then "1" else "0";
+
+  fields = {
+    PID = 1;
+    COMM = 2;
+    STATE = 3;
+    PPID = 4;
+    PGRP = 5;
+    SESSION = 6;
+    TTY_NR = 7;
+    TPGID = 8;
+    MINFLT = 10;
+    MAJFLT = 12;
+    PRIORITY = 18;
+    NICE = 19;
+    STARTTIME = 21;
+    PROCESSOR = 38;
+    M_SIZE = 39;
+    M_RESIDENT = 40;
+    ST_UID = 46;
+    PERCENT_CPU = 47;
+    PERCENT_MEM = 48;
+    USER = 49;
+    TIME = 50;
+    NLWP = 51;
+    TGID = 52;
+    CMINFLT = 11;
+    CMAJFLT = 13;
+    UTIME = 14;
+    STIME = 15;
+    CUTIME = 16;
+    CSTIME = 17;
+    M_SHARE = 41;
+    M_TRS = 42;
+    M_DRS = 43;
+    M_LRS = 44;
+    M_DT = 45;
+    RCHAR = 103;
+    WCHAR = 104;
+    SYSCR = 105;
+    SYSCW = 106;
+    RBYTES = 107;
+    WBYTES = 108;
+    CNCLWB = 109;
+    IO_READ_RATE = 110;
+    IO_WRITE_RATE = 111;
+    IO_RATE = 112;
+    CGROUP = 113;
+    OOM = 114;
+    IO_PRIORITY = 115;
```

Comment:
I'd like these attributes be available when setting a list of them, is something like

```
programs.htop.fields = with htopFields; [ PID COMM ];
```

instead of using strings?

Also these maybe should be renamed to camelCase. Also the numbers should be decreased by one, so the `-1` when writing the config file isn't necessary (this list is directly ripped from the source). Also some attributes may be missing, see [here](https://github.com/hishamhm/htop/blob/master/linux/LinuxProcess.h#L40-L77).